cmd/sync.go

```diff
@@ -111,6 +110,13 @@ func (raw *rawSyncCmdArgs) cook() (cookedSyncCmdArgs, error) {
 		return cooked, fmt.Errorf("source '%s' / destination '%s' combination '%s' not supported for sync command ", raw.src, raw.dst, cooked.fromTo)
 	}
 
+	// Do this check seperately so we don't end up with a bunch of code duplication when new src/dsts are added
+	if cooked.fromTo.From() == common.ELocation.Local() {
```

Comment:
I suspect that the calls to cleanLocalPath, which are in the if/else ladder just above this point, should be taken out of there, and moved to here, so that these lines look like

```
cooked.source = cleanLocalPath(common.ToExtendedPath(cooked.source))
```
Why?  Because 
(a) That makes is consistent with copy, where the conversion to extended path is done befoere the clean and 
(b) That allows the panic-based check inside DeterminePathSeparator, which I've suggested in another comment. (That panic-based check is not usable if cleanLocalPath comes first.  But when ToExtendedPath comes first, it will do an Abs, wihch normalizes the delimiters to the OS default).

Does that sound reasonable?  Any downside to the above?

Reply:
Well, I agree that it makes the code more consistent, and that it drops the code into one place. I think it'd actually be best to write a function called 'cleanLocation' or something of that variety and have it return the source path and SAS token. Then we'd just rail-road it down through that function and only have two calls to it, rather than 20 lines to clean up the source and destination.

@zezha-msft since you've done most of the work on the Sync function, what do you think of that solution?